src/affordance/tasks/arxiv_summarization.py

```diff
@@ -0,0 +1,368 @@
+    labels_list = []
+    runs = 5
+    for run in range(runs):
+        print(f"run {run+1}/{runs}")
+        answers = []
+        for x in tqdm(chunks(inputs, 10)):
+            answers.extend(predict(x))
+        preds = [x.strip() for x in answers]
+        predictions_list.append(preds)
+        labels_list.append(labels)
+    print("Rouge performance")
+    print(rouge.compute(predictions=predictions_list, references=labels_list))
+
+
+def get_few_shot_cot_prompt(task_name: str, description: str, strategy: str) -> str:
+    if strategy == "fixed":
+        return few_shot_cot_prompt
+    elif strategy == "random":
+        return random_tasks(N=6)
+    elif strategy == "similar":
+        return similar_tasks(description, io_pairs, N=6)
+    elif strategy == "similar_auto_decomp":
+        return similar_auto_breakdowns(description, io_pairs, N=6)
+    elif strategy == "llm_similar":
+        return llm_similar_tasks(task_name, description, io_pairs, N=6)
+
+
+def few_shot_cot(temperature=0.3, model_name="text-davinci-002", strategy="fixed"):
+    global few_shot_cot_prompt
+    global few_shot_pot_prompt
+    task_name = "Summarize the given articles"
+    task_description = "In these examples, you are given an Article. Break the input down into subtasks in order to solve the task. You can read and write to an external file memory to help summarize using functions from alexAndKylesAwesomeLibrary."
+
+    prompt = get_few_shot_cot_prompt(task_name, task_description, strategy)
+
+    interpreter = TopDownVisitor(model_name=model_name, temperature=temperature)
+
+    def predict_complete(description, chunk):
+        gpt3 = OpenAIModel(
+            model=model_name, max_length=1000, temperature=temperature, quote="---", n=1
+        )
+        prompts = [few_shot_cot_prompt % (description, x) for x in chunk]
+        outputs = gpt3(prompts)
+        completed_outputs = [
+            interpreter.complete_program(prefix, output) for prefix, output in zip(prompts, outputs)
+        ]
+        return completed_outputs
+
+    predictions_list = []
+    labels_list = []
+    runs = 5
+    for run in range(runs):
+        print("Run %d" % run)
+        answers = []
+        new_labels = [label.split(" ")[0] for label in labels]
+        for x in tqdm(
+            chunks(inputs, 10)
+        ):  # should we replace the chunks function by inputs.split('\n') to use paragraphs ? or use the chunks function with longer chunks?
+            x = [ex.replace("\nA:", "") for ex in x]
+            answers.extend(predict_complete(task_description, x))
+            # time.sleep(10)
+        preds = [get_answer(x) for x in answers]
+        predictions_list.append(preds)
+        labels_list.append(new_labels)
+    print("Rouge performance")
+    print(rouge.compute(predictions=predictions_list, references=labels_list))
+
+
+auto_cot_cleaned_prompt = """(Physics Questions) Answer these high-school-level physics multiple-choice questions.
```

Comment:
I think we should just blow away anything we haven't actually adapted to the arXiv dataset. So let's just stick with whatever `nl_program` uses and get rid of other unused code. We can always bring it in later if we want to use it. Thoughts?